### PR TITLE
add support for time.duration

### LINKED ISSUE #64: supporting time duration as configurable field
Preparation for contribution

## Is your feature request related to a problem? Please describe

Use time.duration as supported configurable field

## Describe the solution

Allow to parse and import key value with time.Duration as type

### PR DESCRIPTION
**Pull request by YafimK**
## Which problem is this PR solving?
Closes #64. Allow usage of time.Duration as field

## Short description of the changes
Mainly adding the time.Duration as cfg type implemented interface.

### CODE REVIEW

File: sync/sync.go [hunk 1]
```diff
@@ -154,6 +155,43 @@ func (s *String) SetString(val string) error {
...
 }
 
+// TimeDuration is Time.Duration type with concurrent access support.
+type TimeDuration struct {
+	rw    sync.RWMutex
+	value time.Duration
+}
+
+// Get returns the internal value.
+func (s *TimeDuration) Get() time.Duration {
```
Comment (mantzas) [inline sync/sync.go:165]: As codecov mentioned maybe we should write some unit tests to cover the methods for TimeDuration.

**Review by mantzas — CHANGES_REQUESTED** [review]

**Review by mantzas — APPROVED** [review]

File: harvester_integration_test.go [hunk 2]
```diff
@@ -23,6 +23,7 @@ var (
 		`INFO: automatically monitoring consul key "harvester/age"`,
 		`INFO: automatically monitoring consul key "harvester/balance"`,
 		`INFO: automatically monitoring consul key "harvester/has-job"`,
+		`INFO: automatically monitoring consul key "harvester/FunTime"`,
```
Comment (c0nstantx) [inline harvester_integration_test.go:26]: Maybe try keeping the keys consistent for readability and not use camel case (`harvester/fun-time`)

**Review by mantzas — DISMISSED** [review]

**Review by c0nstantx — APPROVED** [review]

**Comment by mantzas** [comment]
@YafimK there is a data race in the tests can you check?

**Review by mantzas — APPROVED** [review]

**Review by c0nstantx — DISMISSED** [review]

**Review by Lazaros-Tsigkakos — APPROVED** [review]

**Comment by mantzas** [comment]
@YafimK Sorry letting this fall through the cracks. On last request. Can you also update the Readme to reflect the new data type added?

**Review by mantzas — APPROVED** [review]

**Review by c0nstantx — APPROVED** [review]

**Review by Lazaros-Tsigkakos — APPROVED** [review]

**Pull request merged**